core/transact.go

```diff
@@ -223,15 +222,15 @@ func (h *Handler) finalizeTxWait(ctx context.Context, txTemplate *txbuilder.Temp
 	return nil
 }
 
-func waitForTxInBlock(ctx context.Context, c *protocol.Chain, tx *bc.Tx, height uint64) (uint64, error) {
+func (h *Handler) waitForTxInBlock(ctx context.Context, tx *bc.Tx, height uint64) (uint64, error) {
```

Comment:
It seems like this part is actually not connected to the rest of the PR? Maybe it should be separate (even though it is small)?